app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,16 @@
+class Admin::UsersController < ApplicationController
+  protect_from_forgery with: :exception
+  before_action :restrict_user_by_role
+  VALID_ROLES = ['admin']
+  def index
+  	@users = User.all
+  end	
+  protected
+
+  # redirect if user not logged in or does not have a valid role
+  def restrict_user_by_role
+    unless current_user && VALID_ROLES.include?(current_user.role)
+      redirect_to root_path # change this to your 404 page if needed
+    end
+  end
+end	
```

Comment:
move restrict_user_by_role to admin controller

change name to authorize_admin
def authorize_admin
  return redirect_to root_path unless current_user.admin?
end